CONTRIBUTING.md

```diff
@@ -0,0 +1,57 @@
+# How to contribute to CSMIM
+
+![Steps for filing an issue](docs/contribute-add.svg)
+
+Would you like to see an interface standardized as a CSMIM object type?
+Or do you wish to enhance or change an existing type?
+
+- **Search for existing CSMIM object types first** before requesting to
+  create a new one.
+
+- **Search the issues for existing discussions of your request.** Contribute
+  to an existing discussion if possible, instead of opening a new issue.
+
+- **Open an issue, describe your requirements and label it an enhancement.**
+  If there are existing object types similar to what you need, mention them
+  and explain what is missing for you and how the type should be enhanced.
+  Note that we cannot change types once they are standardized, we can only
+  replace them by newer types, with backwards-compatibility if possible.
+
+Do you have a very concrete proposal for an object type?
+
+- **Usually it is better to open an issue** and describe your requirements
+  instead of directly proposing an object type YAML, because we will have to
+  discuss the rationale of the object type's details in any case.
+
+- **Fork this repository and add the YAML file specifying your object type.**
```

Comment:
while issues may cover multiple objects, maybe we should consider limiting pull requests to single files? also, the file must be placed in the `types` folder. The path will be linked as soon as it is in service.

Reply:
I agree, but where do I add such commentary that is relevant to us but not to external contributors?

Reply:
I was wondering how we can prevent contributors from the work of creating links in the path folder and thought that a remark here would help.

Reply:
Some observations from me:

- Matthias, you seem to say "contributor" for people that belong to the CSMIM WG. But the file CONTRIBUTING.md file is targeted to "externals", or, better, anybody who wants to contribute to CSMIM. The second section only covers our internal process insofar it touches the externals.
- Creating a link only when a type enters service means that, before a type enters service, there is no documentation which path a type is supposed to have.

I suggest that we create new "documentation" issues for our internal process questions. That avoids blocking this PR from merging.